frontend/src/components/Table.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  Pagination,
+  Toolbar,
+  ToolbarContent,
+  ToolbarItem,
+  ToolbarItemVariant,
+} from '@patternfly/react-core';
+import { TableComposable, Thead, Tr, Th, Tbody, Td } from '@patternfly/react-table';
+import React, { useEffect } from 'react';
+import useTableColumnSort, { SortableData } from 'utilities/useTableColumnSort';
+
+type TableProps<DataType> = {
+  data: DataType[];
+  columns: SortableData<DataType>[];
+  Row: (data: DataType) => React.ReactNode;
```

Comment:
You fixed the issue I was thinking about -- hooks not being able to be fired.

```suggestion
  rowRenderer: (data: DataType) => React.ReactNode;
```